Build breaks on FreeBSD/arm64: mismatched types

### Describe the bug

```
error[E0308]: mismatched types
   --> crates/nu-system/src/freebsd.rs:102:23
    |
102 |                 stat: proc.ki_stat,
    |                       ^^^^^^^^^^^^ expected `i8`, found `u8`

error[E0308]: mismatched types
   --> crates/nu-system/src/freebsd.rs:280:13
    |
279 |         match self.stat {
    |               --------- this expression has type `i8`
280 |             libc::SIDL | libc::SRUN => "Running",
    |             ^^^^^^^^^^ expected `i8`, found `u8`

error[E0308]: mismatched types
   --> crates/nu-system/src/freebsd.rs:280:26
    |
279 |         match self.stat {
    |               --------- this expression has type `i8`
280 |             libc::SIDL | libc::SRUN => "Running",
    |                          ^^^^^^^^^^ expected `i8`, found `u8`

error[E0308]: mismatched types
   --> crates/nu-system/src/freebsd.rs:281:13
    |
279 |         match self.stat {
    |               --------- this expression has type `i8`
280 |             libc::SIDL | libc::SRUN => "Running",
281 |             libc::SSLEEP => "Sleeping",
    |             ^^^^^^^^^^^^ expected `i8`, found `u8`

error[E0308]: mismatched types
   --> crates/nu-system/src/freebsd.rs:282:13
    |
279 |         match self.stat {
    |               --------- this expression has type `i8`
...
282 |             libc::SSTOP => "Stopped",
    |             ^^^^^^^^^^^ expected `i8`, found `u8`

error[E0308]: mismatched types
   --> crates/nu-system/src/freebsd.rs:283:13
    |
279 |         match self.stat {
    |               --------- this expression has type `i8`
...
283 |             libc::SWAIT => "Waiting",
    |             ^^^^^^^^^^^ expected `i8`, found `u8`

error[E0308]: mismatched types
   --> crates/nu-system/src/freebsd.rs:284:13
    |
279 |         match self.stat {
    |               --------- this expression has type `i8`
...
284 |             libc::SLOCK => "Locked",
    |             ^^^^^^^^^^^ expected `i8`, found `u8`

error[E0308]: mismatched types
   --> crates/nu-system/src/freebsd.rs:285:13
    |
279 |         match self.stat {
    |               --------- this expression has type `i8`
...
285 |             libc::SZOMB => "Zombie",
    |             ^^^^^^^^^^^ expected `i8`, found `u8`

For more information about this error, try `rustc --explain E0308`.
error: could not compile `nu-system` (lib) due to 8 previous errors

Caused by:
  process didn't exit successfully: `CARGO=/usr/local/bin/cargo CARGO_CRATE_NAME=nu_system CARGO_MANIFEST_DIR=/wrkdirs/usr/ports/shells/nushell/work/nushell-0.94.2/crates/nu-system CARGO_PKG_AUTHORS='The Nushell Project Developers:procs creators' CARGO_PKG_DESCRIPTION='Nushell system querying' CARGO_PKG_HOMEPAGE='' CARGO_PKG_LICENSE=MIT CARGO_PKG_LICENSE_FILE='' CARGO_PKG_NAME=nu-system CARGO_PKG_README='' CARGO_PKG_REPOSITORY='https://github.com/nushell/nushell/tree/main/crates/nu-system' CARGO_PKG_RUST_VERSION='' CARGO_PKG_VERSION=0.94.2 CARGO_PKG_VERSION_MAJOR=0 CARGO_PKG_VERSION_MINOR=94 CARGO_PKG_VERSION_PATCH=2 CARGO_PKG_VERSION_PRE='' LD_LIBRARY_PATH=/wrkdirs/usr/ports/shells/nushell/work/target/release/deps /usr/local/bin/rustc --crate-name nu_system --edition=2021 crates/nu-system/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --crate-type lib --emit=dep-info,metadata,link -C opt-level=s -C linker-plugin-lto -C metadata=72f262b80fa9e5f7 -C extra-filename=-72f262b80fa9e5f7 --out-dir /wrkdirs/usr/ports/shells/nushell/work/target/aarch64-unknown-freebsd/release/deps --target aarch64-unknown-freebsd -C linker=cc -C strip=debuginfo -L dependency=/wrkdirs/usr/ports/shells/nushell/work/target/aarch64-unknown-freebsd/release/deps -L dependency=/wrkdirs/usr/ports/shells/nushell/work/target/release/deps --extern itertools=/wrkdirs/usr/ports/shells/nushell/work/target/aarch64-unknown-freebsd/release/deps/libitertools-1f8af7c3a0bf0095.rmeta --extern libc=/wrkdirs/usr/ports/shells/nushell/work/target/aarch64-unknown-freebsd/release/deps/liblibc-9409b8c0baa85bec.rmeta --extern log=/wrkdirs/usr/ports/shells/nushell/work/target/aarch64-unknown-freebsd/release/deps/liblog-da70274667ce608e.rmeta --extern nix=/wrkdirs/usr/ports/shells/nushell/work/target/aarch64-unknown-freebsd/release/deps/libnix-6b875f5eebf1bfca.rmeta --extern sysinfo=/wrkdirs/usr/ports/shells/nushell/work/target/aarch64-unknown-freebsd/release/deps/libsysinfo-72f30a1d00ab1249.rmeta -C target-cpu= -C link-arg=-fstack-protector-strong` (exit status: 1)
warning: build failed, waiting for other jobs to finish...
```

### How to reproduce

regular build

### Expected behavior

n/a

### Screenshots

_No response_

### Configuration

n/a

### Additional context

Version: 0.94.2
Log: [log](https://pkg-status.freebsd.org/ampere2/data/main-arm64-default/pe0861e7ef8ca_se2ad879d4c6/logs/nushell-0.94.2.log)

--- a/crates/nu-system/src/freebsd.rs
+++ b/crates/nu-system/src/freebsd.rs
@@ -1,6 +1,7 @@
 use itertools::{EitherOrBoth, Itertools};
 use libc::{
-    kinfo_proc, sysctl, CTL_HW, CTL_KERN, KERN_PROC, KERN_PROC_ALL, KERN_PROC_ARGS, TDF_IDLETD,
+    c_char, kinfo_proc, sysctl, CTL_HW, CTL_KERN, KERN_PROC, KERN_PROC_ALL, KERN_PROC_ARGS,
+    TDF_IDLETD,
 };
 use std::{
     ffi::CStr,
@@ -16,7 +17,7 @@ pub struct ProcessInfo {
     pub ppid: i32,
     pub name: String,
     pub argv: Vec<u8>,
-    pub stat: i8,
+    pub stat: c_char,
     pub percent_cpu: f64,
     pub mem_resident: u64, // in bytes
     pub mem_virtual: u64,  // in bytes

